scripts/teamd_increase_retry_count.py

```diff
@@ -0,0 +1,236 @@
+    portchannelTable = Table(configDb, "PORTCHANNEL")
+    return list(portchannelTable.getKeys())
+
+class LacpPacketListenThread(Thread):
+    def __init__(self, port, targetMacAddress, sendReadyEvent):
+        Thread.__init__(self)
+        self.port = port
+        self.targetMacAddress = targetMacAddress
+        self.sendReadyEvent = sendReadyEvent
+        self.detectedNewVersion = False
+
+    def lacpPacketCallback(self, pkt):
+        if pkt["LACPRetryCount"].version == 0xf1:
+            self.detectedNewVersion = True
+        return self.detectedNewVersion
+
+    def run(self):
+        sniff(stop_filter=self.lacpPacketCallback, iface=self.port, filter="ether proto 0x8809 and ether src {}".format(self.targetMacAddress),
+                store=0, timeout=30, started_callback=self.sendReadyEvent.set)
+
+def sendLacpPackets(packets, revertPackets):
+    while not revertTeamdRetryCountChanges:
+        for port, packet in packets:
+            sendp(packet, iface=port)
+        time.sleep(15)
+    if revertTeamdRetryCountChanges:
+        for port, packet in revertPackets:
+            sendp(packet, iface=port)
+
+def abortTeamdChanges(signum, frame):
+    print("Got signal {}, reverting teamd retry count change".format(signum))
+    revertTeamdRetryCountChanges = True
+
+def main(probeOnly=False):
+    if os.geteuid() != 0:
+        print("Root privileges required for this operation")
+        sys.exit(1)
+        return False
+    portChannels = getPortChannels()
+    if not portChannels:
+        return True
+    failedPortChannels = []
+    if probeOnly:
+        for portChannel in portChannels:
+            config = getPortChannelConfig(portChannel)
+            lldpInfo = getLldpNeighbors()
+            portChannelChecked = False
+            for portName in config["ports"].keys():
+                interfaceLldpInfo = [k for k in lldpInfo["lldp"]["interface"] if portName in k]
+                if not interfaceLldpInfo:
+                    print("WARNING: No LLDP info available for {}; skipping".format(portName))
+                    continue
+                interfaceLldpInfo = interfaceLldpInfo[0][portName]
+                peerName = list(interfaceLldpInfo["chassis"].keys())[0]
+                peerInfo = interfaceLldpInfo["chassis"][peerName]
+                if "descr" not in peerInfo:
+                    print("WARNING: No peer description available via LLDP for {}; skipping".format(portName))
+                    continue
+                portChannelChecked = True
+                if "SONiC" not in peerInfo["descr"]:
+                    print("WARNING: Peer device is not a SONiC device; skipping")
+                    failedPortChannels.append(portChannel)
+                    break
+
+                sendReadyEvent = Event()
+
+                # Start sniffing thread
+                lacpThread = LacpPacketListenThread(portName, config["ports"][portName]["runner"]["partner_lacpdu_info"]["system"], sendReadyEvent)
```

Comment:
This access of config can potentially result in KeyError

Reply:
Added some checks to verify that at least some level of keys are returned from `teamd`.